content/en/docs/Monitoring/opentelemetry-integration.md

```diff
@@ -0,0 +1,272 @@
+    receivers:
+      otlp:
+        protocols: 
+          grpc:
+            endpoint: ":8000"
+            tls:
+              cert_file: /etc/ssl/certs/server/server.pem
+              key_file: /etc/ssl/certs/server/server-key.pem
+              ca_file: /etc/ssl/certs/ca/ca.pem
+    processors:
+      batch:
+        send_batch_size: 10000
+        timeout: 5s
+    extensions:
+      health_check: {}
+    exporters:
+      jaeger:
+        endpoint: "jaeger-collector.observability.svc.cluster.local:14250"
+        tls:
+          insecure: true
+      prometheus:
+        endpoint: ":9090"
+      logging:
+        loglevel: debug
+    service:
+      extensions: [health_check]
+      pipelines:
+        traces:
+          receivers: [otlp]
+          processors: []
+          exporters: [jaeger, logging]
+        metrics:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [prometheus, logging]
+```
+
+**deployment.yaml**
+
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: opentelemetrycollector
+  namespace: kyverno
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: opentelemetrycollector
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: opentelemetrycollector
+    spec:
+      containers:
+      - name: otelcol
+        args:
+        - --config=/conf/collector.yaml
+        image: otel/opentelemetry-collector:0.50.0
+        volumeMounts:
+        - name: collector-config
+          mountPath: /conf
+        - name: otel-collector-secrets
+          mountPath: /etc/ssl/certs/server
+        - name: root-ca
+          mountPath: /etc/ssl/certs/ca
+      volumes:
+      - configMap:
+          name: collector-config
+          items:
+          - key: collector.yaml
+            path: collector.yaml
+        name: collector-config
+      - secret:
+          secretName: otel-collector-secrets
+        name: otel-collector-secrets
+      - secret:
+          secretName: root-ca
+        name: root-ca
+```
+
+This will ensure that Opentelemetry Collector only accept encrypted data on the receiver endpoint. 
+
+Pass the flag ```transportCreds``` as the secret name containing the "ca.pem" file (Empty string means insecure connection will be used).
+
+### Setting up Prometheus
+
+- For the metrics backend you can install Prometheus on you cluster. For a general example we have a ready-made configuration for you. Install Promethues by running:
+
+```kubectl apply -k github.com/kyverno/grafana-dashboard/examples/prometheus```
+
+- Port-forward the Prometheus service to view the metrics on localhost.
+
+
+### Setting up Jaeger
+
+The traces are pushed to the Jaeger backend on port **14250**. To install Jaeger:
+
+- First install the Jaeger Operator
+
+```shell
+kubectl create namespace observability
+kubectl create -n observability -f https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml
```

Comment:
Got the following error when running this command:
```
➜  ~ kubectl create -n observability -f https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml
customresourcedefinition.apiextensions.k8s.io/jaegers.jaegertracing.io created
serviceaccount/jaeger-operator created
role.rbac.authorization.k8s.io/leader-election-role created
role.rbac.authorization.k8s.io/prometheus created
clusterrole.rbac.authorization.k8s.io/jaeger-operator-metrics-reader created
clusterrole.rbac.authorization.k8s.io/manager-role created
clusterrole.rbac.authorization.k8s.io/proxy-role created
rolebinding.rbac.authorization.k8s.io/leader-election-rolebinding created
rolebinding.rbac.authorization.k8s.io/prometheus created
clusterrolebinding.rbac.authorization.k8s.io/jaeger-operator-proxy-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/manager-rolebinding created
service/jaeger-operator-metrics created
service/jaeger-operator-webhook-service created
deployment.apps/jaeger-operator created
mutatingwebhookconfiguration.admissionregistration.k8s.io/jaeger-operator-mutating-webhook-configuration created
validatingwebhookconfiguration.admissionregistration.k8s.io/jaeger-operator-validating-webhook-configuration created
unable to recognize "https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml": no matches for kind "Certificate" in version "cert-manager.io/v1"
unable to recognize "https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml": no matches for kind "Issuer" in version "cert-manager.io/v1"
```

Reply:
Cert manager is needed for the installation. I will add this to the docs